webapp/src/js/enketo/widgets/dynamic-url.js

```diff
@@ -17,7 +17,20 @@ class DynamicUrlWidget extends Widget {
     const urlElement = currentElement.find('.url');
     const setHref = () => currentElement.attr('href', urlElement.text());
     setHref();
-    urlElement.on('DOMSubtreeModified', setHref);
+
+    const observer = new MutationObserver(mutationList => {
+      mutationList.forEach(mutation => {
+        if (mutation.type === 'childList' || mutation.type === 'characterData') {
```

Comment:
Minor, but do we actually need this check? (I am not a MutationObserver expert, so I am trying to understand what all might be happening here!)

It seems like since we are locking down the observations below to only have `childList: true` and `characterData: true`, we should not need to filter anything here.  Any time the observer picks up a change, we should call `setHref`.  (And testing this locally, I only saw `childList` mutations coming through here).

Reply:
From my testing, and as you've noted, the check seems to be a bit redundant and no `characterData` mutation is observed.

For future further reading:
https://stackoverflow.com/a/10719879
https://lists.w3.org/Archives/Public/public-webapps/2011JulSep/1622.html
https://davidwalsh.name/mutationobserver-api (MutationObserver options)
https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver/observe#removed_descendants_when_using_subtree
https://developer.mozilla.org/en-US/docs/Web/API/MutationRecord/target#logging_the_target_of_a_mutation